### PR TITLE
[WIP] Add support for SpaCy text classification models

### PR DESCRIPTION
**Pull request by suzil**
Refs: https://github.com/mlflow/mlflow/issues/500

### CODE REVIEW

File: examples/spacy/text_classification.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from pathlib import Path
+import thinc.extra.datasets
+
+import mlflow
+import mlflow.spacy
+import spacy
+from spacy.util import minibatch, compounding
+
+
+@plac.annotations(
```
Comment (mparkhe) [inline examples/spacy/text_classification.py]: Can we use `click` for this?

Here are a couple of references:
1. [cli.py](https://github.com/mlflow/mlflow/blob/master/mlflow/cli.py)
2. [hyperparam/train.py](https://github.com/mlflow/mlflow/blob/master/examples/hyperparam/train.py#L111)

**Review by mparkhe — CHANGES_REQUESTED** [review]
Hi @suzil : Thanks for this pull request. There are some lint errors. Also made a couple of change requests, one of them should help resolve most of the lint issues.

File: examples/spacy/text_classification.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                mlflow.log_metric("precision", scores["textcat_p"])
+                mlflow.log_metric("recall", scores["textcat_r"])
+                mlflow.log_metric("f-score", scores["textcat_f"])
+
+            # Test the trained model
+            test_text = "This movie sucked"
+            doc = nlp(test_text)
+            print(test_text, doc.cats)
+
+            mlflow.spacy.log_model(nlp, "model")
```
Comment (mparkhe) [inline examples/spacy/text_classification.py]: Please extend this example to load this saved model and predict.
alternatively, add a unit test to cover the log_model and load_model flow.

**Review by mparkhe — COMMENTED** [review]
Hi @suzil : couple of small comments. I tried out the code -- it looks cool. Thanks for adding this flavor and example.

**Comment by suzil** [comment]
Hi @mparkhe. Thanks so much for the feedback.

I have a question: how should I handle passing a ["mode" parameter](https://github.com/mlflow/mlflow/pull/610/files#diff-fcf5da7aba04355ed78ee4c3c04a9b40R128) to specify the kind of model SpaCy should make predictions on? I handle it in the code by having the user pass an extra blank row that includes a column called "mode", but this feels very kludgey.

**Comment by tomasatdatabricks** [comment]
Hi @suzil. That is a very good question. 

I would recommend storing the desired mode with the model - i.e. you add the argument to mlflow.spacy.save_model and store it with the model data. The produced pyfunc would then only score the one specified mode. This has a disadvantage that you need to produce multiple copies o the same model if you need to score different modes as pyfunc but it is imho better than encoding mode with the data and I don't see other good way around it.

**Comment by suzil** [comment]
Closing due to staleness.

**Comment by TheTravellingSalesman** [comment]
Is the branch for this PR still available somewhere? I'm interested in picking up this development.

**Pull request closed without merging**